### ISSUE TITLE
[Bug]: Incoming audio sounds as if it's been severly slowed down

### ISSUE THREAD

**Issue by tejas-n** [BelledonneCommunications/linphone-sdk#243]
### Context

The Linphone app has always run smoothly on our custom hardware running Android 7.  Recently we tried running it on our newer model which has a different audio card and which runs Android 10, but we've ran across a weird issue where the incoming audio sounds like it's been slowed down severely and mostly sounds garbled. The person on the other end can hear fine though.

We mainly use it to connect to the Twilio SIP servers but we could replicate this issue with sip.linphone.org as well. I've attached the link to the log as well.

### General information

- Device: Custom hardware
- OS: Android 10
- Version of the App: 4.6.11
- Version of the SDK: 5.1.46

### Expected behaviour

Clear audio is heard when the call is established.

### To Reproduce

1. Make an outgoing call
2. After the call connects, the other person's voice sounds like it's being slowed down severely and is mostly garbled.
3. The person on the other end can hear the audio fine though

### SDK logs URL

https://www.linphone.org:444//tmp/62ba464775bef_c56ed1b434162e035737.gz

**Comment by tejas-n** [comment]
Thanks for the quick response, @Viish. I tried toggling early media and tried the other audio card as well but it doesn't seem to help. Unfortunately we can't use Opus because Twilio SIP servers only support PCMU and PCMA.

We tried using Skype and Zoom just to see if it's a hardware problem but they seemed to have worked. Just to give you an idea of what it sounds like, I've uploaded the Linphone recording to https://drive.google.com/file/d/1QbMUWVQuVA30Xgfgx-CefMk4pUKsQbzF/view?usp=sharing

You can hear me saying "hello" in that call.

**Comment by tejas-n** [comment]
That seems to have fixed the issue for some reason. Thanks.

**Comment by Viish** [comment]
Yes that's because of this error, caused by your Android firmware badly supporting TelecomManager APIs:
```
2022-06-28 10:06:40:426 [org.linphone/linphone-android] ERROR [Telecom Helper] Exception while trying to place call: java.lang.SecurityException: placeCall: Neither user 10094 nor current process has android.permission.CALL_PHONE.
```
When we rely on TelecomManager, we don't touch the AudioManager as it is already done by TelecomManager, but in your case it isn't, so audio isn't tuned for voice communications.

**Comment by tejas-n** [comment]
Hi @Viish. Following your last comment, I think I found out why it's happening. I think it's a Linphone Android app issue rather than the SDK. Our devices don't support `ConnectionService`, so this returns false on our devices:

`packageManager.hasSystemFeature(PackageManager.FEATURE_CONNECTION_SERVICE)`

If you look at the DialerFragment source code
https://github.com/BelledonneCommunications/linphone-android/blob/9d082e2630e8ecf1e3145ca12e39b0b633cf644a/app/src/main/java/org/linphone/activities/main/dialer/fragments/DialerFragment.kt#L278-L291

It calls `corePreferences.useTelecomManager = true` even when it detects that the device doesn't support connection service. If I change it to the code below, the calls work fine

```
private fun enableTelecomManager() {
        Log.i("[Dialer] Telecom Manager permissions granted")
        if (!TelecomHelper.exists()) {
            Log.i("[Dialer] Creating Telecom Helper")
            if (requireContext().packageManager.hasSystemFeature(PackageManager.FEATURE_CONNECTION_SERVICE)) {
                TelecomHelper.create(requireContext())
                corePreferences.useTelecomManager = true
            } else {
                Log.e("[Dialer] Telecom Helper can't be created, device doesn't support connection service")
            }
        } else {
            Log.e("[Dialer] Telecom Manager was already created ?!")
            corePreferences.useTelecomManager = true
        }
    }
 ```
 
 I'm not sure if the `corePreferences.useTelecomManager = true`  is called elsewhere(even after detecting that `ConnectionService` is not supported). What are your thoughts?

**Comment by Viish** [comment]
Thanks for noticing the issue! I'll fix it right away.

**Comment by tejas-n** [comment]
Thanks.
